packages/react/src/migrations/update-19-6-0/update-ssr-server-port.ts

```diff
@@ -0,0 +1,132 @@
+          );
+          if (serverContent && port) {
+            const updatedServerContent = updateServerPort(serverContent, port);
+            if (updatedServerContent) {
+              tree.write(
+                joinPathFragments(project.root, 'server.ts'),
+                updatedServerContent
+              );
+              break;
+            }
+          }
+        }
+      }
+    }
+  }
+}
+
+function updateServerPort(serverContent: string, port: number) {
+  const sourceFile = tsquery.ast(serverContent);
+
+  const serverPortNode = tsquery(
+    sourceFile,
+    `VariableDeclaration:has(Identifier[name="port"])`
+  )[0];
+  if (serverPortNode) {
+    const binaryExpression = tsquery(serverPortNode, 'BinaryExpression')[0];
+    if (binaryExpression) {
+      const leftExpression = tsquery(
+        binaryExpression,
+        'PropertyAccessExpression:has(Identifier[name="env"])'
+      )[0];
+      const rightExpression = tsquery(
+        binaryExpression,
+        'NumericLiteral[text="4200"]'
+      )[0];
+
+      if (leftExpression && rightExpression) {
+        const serverPortDeclaration = serverPortNode as ts.VariableDeclaration;
+        const newInitializer = ts.factory.createNumericLiteral(port.toString());
+
+        const updatePortDeclaration = ts.factory.updateVariableDeclaration(
+          serverPortDeclaration,
+          serverPortDeclaration.name,
+          serverPortDeclaration.exclamationToken,
+          serverPortDeclaration.type,
+          newInitializer
+        );
+
+        const updatedStatements = sourceFile.statements.map((statement) => {
+          if (ts.isVariableStatement(statement)) {
+            const updatedDeclarationList =
+              statement.declarationList.declarations.map((decl) =>
+                decl === serverPortDeclaration ? updatePortDeclaration : decl
+              );
+
+            const updatedDeclList = ts.factory.updateVariableDeclarationList(
+              statement.declarationList,
+              updatedDeclarationList
+            );
+
+            return ts.factory.updateVariableStatement(
+              statement,
+              statement.modifiers,
+              updatedDeclList
+            );
+          }
+
+          return statement;
+        });
+
+        const updatedSourceFile = ts.factory.updateSourceFile(
+          sourceFile,
+          updatedStatements
+        );
+
+        const printer = ts.createPrinter();
+        return printer.printNode(
+          ts.EmitHint.Unspecified,
+          updatedSourceFile,
+          sourceFile
+        );
+      }
+    }
+  }
+}
+
+function isModuleFederationSSRProject(
+  tree: Tree,
+  project: ProjectConfiguration
+) {
+  let hasMfeServerConfig = false;
+  let hasServerFile = false;
+  visitNotIgnoredFiles(tree, project.root, (filePath) => {
+    if (minimatch(filePath, '**/module-federation*.server.config.*')) {
+      hasMfeServerConfig = true;
+    }
+    if (minimatch(filePath, '**/server.ts')) {
```

Comment:
I don't think we need `minimatch` here since later we're only handling `<project-root>/server.ts`.